crates/prune/src/parts/receipts.rs

```diff
@@ -0,0 +1,186 @@
+use crate::{
+    parts::{Part, PruneInput, PruneOutput, PruneOutputCheckpoint},
+    PrunerError,
+};
+use reth_db::{database::Database, tables};
+use reth_interfaces::RethResult;
+use reth_primitives::{PruneCheckpoint, PrunePart};
+use reth_provider::{DatabaseProviderRW, PruneCheckpointWriter, TransactionsProvider};
+use tracing::{instrument, trace};
+
+pub(crate) struct Receipts {}
```

Comment:
```suggestion
#[derive(Default)]
#[non_exhaustive]
pub(crate) struct Receipts {}
```

Reply:
added both and started using `::default()`, but since we use these structs from the same crate, `non_exhaustive` will not prevent the construction of struct via struct expression